### ISSUE TITLE
False positive undefined-loop-variable with pylint 2.5.3

### ISSUE THREAD

**Issue by lummax** [PyCQA/pylint#3711]
### Steps to reproduce
1. Save code as `false-positive.py`
2. Run `python3 -m pylint --disable=all --enable=undefined-loop-variable false-positive.py`

```python
def test(content):
    def handle_line(line):
        if "X" in line:
            line = line.replace("X", "Y")
        elif "Y" in line:  # line 5
            line = '{}'.format(line)
        elif "Z" in line:  # line 7
            line = f'{line}'
        else:
            line = '%s' % line  # line 10

    for line in content.split('\n'):
        handle_line(line)
```

### Current behavior

This prints:

```
false-positive.py:5:20: W0631: Using possibly undefined loop variable 'line' (undefined-loop-variable)
false-positive.py:7:20: W0631: Using possibly undefined loop variable 'line' (undefined-loop-variable)
false-positive.py:10:26: W0631: Using possibly undefined loop variable 'line' (undefined-loop-variable)
```

### Expected behavior

No `undefined-loop-variable` violations.

### pylint --version output

```
pylint 2.5.3
astroid 2.4.2
Python 3.7.7 (default, Mar 10 2020, 06:34:06) 
[GCC 9.3.0]
```

**Comment by bodograumann** [comment]
I am seeing a similar issue here:
```python
lst = []
lst2 = [1, 2, 3]

for item in lst:
    pass

bigger = [
    [
        x for x in lst2 if x > item
    ]
    for item in lst
]
```

pylint complains:
> pylint-test.py:9:31: W0631: Using possibly undefined loop variable 'item' (undefined-loop-variable)

Removing the first `for` loop makes it vanish.
When pylint determines that `lst` definitely contains items, the warning also disappears, but it does not apply even for an empty `lst`, because then the outer list comprehension is simply empty as well.
